lib/publiccloud/provider.pm

```diff
@@ -719,7 +719,8 @@ To get the complete output structure, the call is:
 sub get_terraform_output {
     my ($self, $jq_query) = @_;
     my $res = script_output("terraform output -json | jq -r '$jq_query' 2>/dev/null", proceed_on_failure => 1);
-    return $res;
+    # jq 'null' shall return empty
+    return $res unless ($res =~ /null/);
```

Comment:
```suggestion
    return $res unless ($res =~ /^null$/);
```
What if you get something like:
```json
{
  "Instance": "my-fancy-vm",
  "disk-1": "opensuse.qcow2",
  "disk-2": "null"
}
```

Reply:
Do you mean: return an hash, containing more keys?

That routine generally returns a json structure,  defined by the input query that he/her uses. But more often the expected type is a basic type like a string or integer.

So it should be up to the caller handle the return value like a pointer (to and hash) or already a final value.

Reply:
Moreover, I would advise _against_ using `/^null$/`, because by experience from VR tests that I ran, in logs I noted often in such 
cases _extra characters_ added before and after `null`, related to some visual formatting, using escaped sequences, (somethings like `^a1b`**null**`c3d`).
Therefore I preferred to leave the search to work "internally" to the effective string

Reply:
well, after our clarification, indeed if output is a terraform structure containing some valid data and `null` too, that expression will return no data.
I will deeper check it.

Reply:
Applied `terraform ... -no-color` option to filter extra char. and strict string matching.

Reply:
done